biogeochem/EDPhysiologyMod.F90

```diff
@@ -1106,229 +1165,159 @@ subroutine phenology_leafonoff(currentSite)
+                write(fates_log(),'(a,1x,i5)'   ) ' Stress deciduous = ',prt_params%stress_decid(ipft)
+                write(fates_log(),'(a,1x,i5)'   ) ' Site status      = ',currentSite%cstatus
+                write(fates_log(),'(a,1x,i5)'   ) ' Cohort status    = ',currentCohort%status_coh
+                write(fates_log(),'(a,1x,f12.5)') ' DBH              = ',currentCohort%dbh
+                write(fates_log(),'(a,1x,f12.5)') ' Leaf_c           = ',leaf_c
+                write(fates_log(),'(a,1x,f12.5)') ' Store_c          = ',store_c
+                write(fates_log(),'(a,1x,f12.5)') ' Fnrt_c           = ',fnrt_c
+                write(fates_log(),'(a,1x,f12.5)') ' Sapw_c           = ',sapw_c
+                write(fates_log(),'(a,1x,f12.5)') ' Struct_c         = ',struct_c
+                write(fates_log(),'(a)') '---------------------------------------------'
+                write(fates_log(),'(a)') ''
+                write(fates_log(),'(a)') ''
+                call endrun(msg=errMsg(sourcefile, __LINE__))
+             end select
+          end if
 
-                      call PRTDeciduousTurnover(currentCohort%prt,ipft, &
-                           sapw_organ, stem_drop_fraction)
 
-                      call PRTDeciduousTurnover(currentCohort%prt,ipft, &
-                           struct_organ, stem_drop_fraction)
-                   endif
+          ! MLO. To avoid duplicating code for drought and cold deciduous PFTs, we first
+          !      check whether or not it's time to flush or time to shed leaves, then
+          !      use a common code for flushing or shedding leaves.
+          if (prt_params%season_decid(ipft) == itrue) then ! Cold deciduous
+             ! A. Is this the time for COLD LEAVES to switch to ON?
+             is_flushing_time = ( currentSite%cstatus      == phen_cstat_notcold .and. & ! We just moved to leaves being on
+                                  currentCohort%status_coh == leaves_off         )        ! Leaves are currently off
+             ! B. Is this the time for COLD LEAVES to switch to OFF?
+             is_shedding_time = ( currentSite%cstatus == phen_cstat_nevercold  .or.  & ! Past leaf drop day or...
+                                  currentSite%cstatus == phen_cstat_iscold   ) .and. & ! Too cold?
+                                currentCohort%status_coh == leaves_on          .and. & ! Leaves have not dropped yet
+                                ( currentCohort%dbh > EDPftvarcon_inst%phen_cold_size_threshold(ipft) .or. & ! Grasses are big enough or...
+                                  prt_params%woody(ipft) == itrue                                     )      ! this is a woody PFT.
+
+          elseif (prt_params%stress_decid(ipft) == itrue ) then ! Drought deciduous
+             ! A. Is this the time for DROUGHT LEAVES to switch to ON?
+             is_flushing_time = ( currentSite%dstatus(ipft) == phen_dstat_moiston  .or.  & ! Conditions are sufficiently moist
```

Comment:
While this is being overhauled, is there an argument for making leaves come on over some period of time rather than just on one day? Not that it's 100% necessary, just thought I'd mention it...

Reply:
I think this will be one of the options I am going to try (along with other approaches to decide abscission and flushing times).  One way of doing it would be similar to ED-2.2, where we have two moisture thresholds. The maximum leaf biomass goes from zero at the lower threshold to one at the upper threshold.